### ISSUE TITLE
eruby filetype plugin leaks memory

### ISSUE THREAD

**Issue by dunrix** [vim-ruby/vim-ruby#431]
### Invoking filetype detection on eruby buffer results in memory leaking.

Way to reproduce:

 1. open an eruby file in vim buffer, like `test.erb` with following contents
```
<!DOCTYPE html>
<html>
  <body>
  </body>
</html>
```
 2. make sure filetype plugin is enabled
`filetype plugin on`
 3. check vim process resident memory size. Around 21 MB in my case. 
 4. invoke `filetype detect` command many times to observe memory grow. I did it with
```
:for i in range(20)
:  filetype detect
:endfor
```
 5. check vim process memory again to see the leak. ~ 1.1 GB in my case.

Additional notes:
 - it leaks only when `filetype plugin` is on. It may be related to `filetype detect` behavior, quoting from vim docs for *filetype-detect*:
 >    When filetype detection was off, it will be enabled first, like the "on"
argument was used.
 - it does not leak in case of other file types like ruby, python, html. Observed only at *eruby*.
 - Vim  8.2.2891 on x86_64 Linux
 - eruby ftplugin - last change 2019 Jan 06
 
I'm sorry for previous x-report on vim-rails. Now I've been finally able to narrow-down environment state at which reported issue reliably occurs.

**Comment by tpope** [comment]
Unlike with rails.vim, this should be easy enough to narrow down yourself. Start deleting stuff from the bottom of `ftplugin/eruby.vim` and keep going until the issue goes away. Just make sure you keep `let b:did_ftplugin = 1` so that another version of the ftplugin doesn't load.

**Comment by dunrix** [comment]
Ok, that was fast, although I'm not learned in vimscript.

The cause of leak was improperly set `b:undo_ftplugin` for unloading stuff, by omitting concatenation operator.

Found out this bug was already [fixed](https://github.com/vim-ruby/vim-ruby/commit/416be2b00de6390f05da2f60f18aa15a4397eef7) year ago in your repo, however Vim [upstream](https://github.com/vim/vim/blob/4a27aefe3b5652ee58b2eb85e9d09b542b087aaa/runtime/ftplugin/eruby.vim#L121-L122) is still out of sync.

Vim issue reporting policy suggests contact authors directly in case of plugins. That's why I came here… :confused:
